euphonic_wrapper.py

```diff
@@ -0,0 +1,277 @@
+        self.lim = np.inf
+        self.scattering_lengths = 'Sears1992'
+        self.weights = None
+        self.asr = None
+        self.dipole = True
+        self.eta_scale = 1.0
+        self.splitting = True
+        self.insert_gamma = False
+        self.reduce_qpts = True
+        self.use_c = False
+        self.n_threads = 1
+        self.fall_back_on_python = True
+        
+        if value is not None:
+            if isinstance(value, str):
+                self.load(value)
+            elif hasattr('calculate_qpoint_phonon_modes'):
+                self.force_constants = value
+            elif hasattr('calculate_structure_factor'):
+                self.phonon_modes = value
+            elif isinstance(value, dict):
+                kwargs.update(value)
+
+        # Simple parameters can be set in keyword arguments
+        if len(kwargs.keys()) > 0:
+            for ky in ['debye_waller_grid', 'bose', 'negative_e', 'conversion_mat', 'chunk', 'lim', 'weights', 'asr',
+                       'dipole', 'eta_scale', 'splitting', 'insert_gamma', 'reduce_qpts', 'use_c', 'n_threads',
+                       'fall_back_on_python', 'temperature']:
+                if ky in kwargs:
+                    setattr(self, ky, kwargs.pop(ky))
+
+        # If there are still keyword arguments they should relate to constructing ForceConstants or QpointPhononModes
+        if len(kwargs.keys()) > 0:
+            if 'path' in kwargs:
+                try:
+                    self.force_constants = self.from_phonopy(kwargs)
+                except:
+                    self.from_dict(kwargs)
+            else:
+                self.from_dict(kwargs)
+                
+    def load(self, value):
+        # Tries to automatically determine type of input and auto-load it
+        if os.path.isdir(value):
+            self.force_constants = ForceConstants.from_phonopy(path=value)
+        elif os.path.isfile(value):
+            with open(value, 'rb') as fil:
+                byt = fil.read(14)
+            if byt == b'\x00\x00\x00\nCASTEP_BIN':
+                self.force_constants = ForceConstants.from_castep(value)
+            elif byt[:9] == b'phonopy:\n':
+                self.force_constants = ForceConstants.from_phonopy(path=os.path.dirname(value), summary_name=value)
+            elif byt[:8] == b'\x89HDF\r\n\x1a\n':
+                # Assumes that the summary is in the 'phonopy.yaml' file
+                self.force_constants = ForceConstants.from_phonopy(path=os.path.dirname(value), fc_name=value)
+            elif byt[:2] == b'{\n':
+                self.force_constants = ForceConstants.from_json_file(value)
+            elif byt == b' BEGIN header\n':
+                self.phonon_modes = QpointPhononModes.from_castep(value)
+            else:
+                raise RuntimeError(f'Cannot read {filename} due to unknown format')
+        else:
+            raise RuntimeError('Input string must be a file or folder name')
```

Comment:
Trying to guess the input type like this could be error prone, maybe we should make the user be more explicit? See the main review text

Reply:
Yeah... it might be fragile... I think the checks for the `.castep_bin` and `hdf5` files is robust (I think those files always starts with those magic bytes) For the `yaml` and `json` files it's not so robust. 

The issue is that the implementation of the wrapper class uses `subsref` which only works for methods within an object - e.g. for class (static) methods like `from_castep` and `from_phonopy` we can't use `subsref` to access them from Matlab - we'd have to define them in the Matlab wrapper explicitly. This is of course no reason not to do it, but for the demo/prototype I just wanted to minimise the Matlab code you'd have to write to wrap a Python class.

Reply:
Removed this from the code - users have to construct the `ForceConstants` object explicitly with `from_castep`, `from_phonopy` etc. (or with `dicts`). They can then construct a `EuphonicWrapper` using the obtained `ForceConstants` (or other object to be determined in future.